tests/integration/workflows/test_matching.py

```diff
@@ -167,3 +170,82 @@ def test_stop_matched_holdingpen_wfs(app, simple_record):
+    }
+
+    obj = workflow_object_class.create(
+        data=record,
+        id_user=1,
+        data_type='hep'
+    )
+    eng = object()
+
+    assert fuzzy_match(obj, eng)
+    assert 'matches' in obj.extra_data
+
+    matches = get_value(obj.extra_data, 'matches.fuzzy')
+
+    # assert the result with correct abstract comes as a first result
+    assert matches[0]['control_number'] == record_1.json['control_number']
```

Comment:
looks great :+1: 
could you also assert that you're not mutating the abstract? that is, before running the fuzzy matcher on L245 you do
```
original_abstract = obj.data["abstracts"][0]["value"]
```
then at the end you add
```
assert obj.data["abstracts"][0]["value"] == original_abstract
```
The reason I'm asking is that I have a strong suspicion that this will fail, because on L94 of the code you do a shallow copy of `obj.data`, but then you reassign `abstract["value"]` which is still shared with the original `obj.data`.